lib/trino-filesystem/src/test/java/io/trino/filesystem/AbstractTestTrinoFileSystem.java

```diff
@@ -141,19 +158,19 @@ void testInputFileMetadata()
             throws IOException
     {
         // an input file cannot be created at the root of the file system
-        assertThatThrownBy(() -> getFileSystem().newInputFile(getRootLocation()))
+        assertThatThrownBy(() -> getTestingFileSystem().newInputFile(getRootLocation()))
```

Comment:
Having both `getFileSystem()` and `getTestingFileSystem()` is confusing. It will be easy to use the wrong one in the test code. Let's find a better way.

I think we could do the wrapping in the implementations, which would also avoid all the changes in `AbstractTestS3FileSystem`.